src/pages/Account/Venue/VenueWizard.tsx

```diff
@@ -32,12 +43,58 @@ const reducer = (
   switch (action.type) {
     case "SUBMIT_TEMPLATE_PAGE":
       return { ...state, templatePage: { template: action.payload } };
+    case "SUBMIT_DETAILS_PAGE":
+      return { ...state, detailsPage: { venue: action.payload } };
     default:
       throw new Error();
   }
 };
 
 export const VenueWizard: React.FC = () => {
+  const { venueId } = useParams();
+
+  return venueId ? (
+    <VenueWizardEdit venueId={venueId} />
+  ) : (
+    <VenueWizardCreate />
+  );
+};
+
+interface VenueWizardEditProps {
+  venueId: string;
+}
+
+const VenueWizardEdit: React.FC<VenueWizardEditProps> = ({ venueId }) => {
+  // get the venue
+  const firestore = useFirestore();
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    const f = async () => {
```

Comment:
I guess you defined this functions because the useEffect function cannot be async. However I would prefer to have a better naming. Perhaps sumitVenueData